ndcube/ndcube.py

```diff
@@ -629,6 +630,74 @@ def explode_along_axis(self, axis):
+        """
+        Reprojects this NDCube to the coordinates described by another WCS object.
+
+        Parameters
+        ----------
+        target_wcs : `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
+            The WCS object on which the ``NDCube`` is to be reprojected.
+
+        shape_out: `tuple`, optional
+            The shape of the output data. If not specified, `~astropy.wcs.wcsapi.BaseLowLevelWCS.pixel_shape` attribute (if available)
+            from the low level API of the ``target_wcs`` is used.
+
+        order: `int` or `str`
+            The order of the interpolation. This can be any of: 'nearest-neighbour', 'bilinear',
+            'biquadratic', or 'bicubic'.
+
+        output_array: `numpy.ndarray`, optional
+            An array in which to store the reprojected data. This can be any numpy array
+            including a memory map, which may be helpful when dealing with extremely large files.
+
+        return_footprint: `bool`
+            Whether to return the footprint in addition to the output array.
+
+        Returns
+        -------
+        resampled_cube : `ndcube.NDCube`
+            A new resultant NDCube object, the supplied ``target_wcs`` will be the ``.wcs`` attribute of the output ``NDCube``.
+
+        footprint: `numpy.ndarray`
+            Footprint of the input array in the output array. Values of 0 indicate no coverage or
+            valid values in the input image, while values of 1 indicate valid values.
+
+        Notes
+        -----
+        This method doesn't support handling of the ``mask``, ``extra_coords``, and ``uncertainty`` attributes yet.
+        However, ``meta`` and ``global_coords`` are copied to the output ``NDCube``.
+        """
+
+        if not utils.wcs.compare_wcs_physical_types(self.wcs, target_wcs):
+            raise Exception('Given target_wcs is not compatible with this NDCube, the physical types do not match.')
+
+        low_level_target_wcs = utils.wcs.get_low_level_wcs(target_wcs, 'target_wcs')
+
+        # If shape_out is not specified explicity, try to extract it from the low level WCS
+        if not shape_out:
+            if hasattr(low_level_target_wcs, 'pixel_shape') and low_level_target_wcs.pixel_shape is not None:
+                shape_out = low_level_target_wcs.pixel_shape
+            else:
+                raise Exception("shape_out must be specified if target_wcs's low level API "
+                                "does not have the pixel_shape attribute.")
+
+        data = reproject_interp(self, output_projection=target_wcs,
+                                shape_out=shape_out, order=order,
+                                output_array=output_array,
+                                return_footprint=return_footprint)
+
+        if return_footprint:
+            data, footprint = data
+
+        resampled_cube = NDCube(data, wcs=target_wcs, meta=deepcopy(self.meta))
```

Comment:
We should return the type of ourselves so that when people subclass us they get the class back they are expecting.
```suggestion
        resampled_cube = type(self)(data, wcs=target_wcs, meta=deepcopy(self.meta))
```